Initialise Express & React , Add routes, Add styling

mattegan111 opened the pull request.

mattegan111 committed:
Initialise Express & React

mattegan111 committed:
Add initial routes

mattegan111 committed:
Adds styling

nathanrobb commented on client/package.json:
```diff
@@ -0,0 +1,42 @@
+{
+  "name": "client",
```
Would this not be Fakebook?

EDIT: So there is a Fakebook `package.json` down below. What is the client folder? Why does this have a `package.json` too?

EDIT2: Okay, I get it now. This is the `react` front end SPA. The other `package.json` is for the `express` server. Perhaps this `package.json` should have more of the fakebook details in it (e.g. `fakebook-app`) and the other file have more server oriented info (e.g. `fakebook-api`)? Also should the server files be in a `server` folder? Currently it's a bit hard to figure out what's going on at a glance.

mattegan111 replied:
Created a 'server' folder and added relevant files and folders. Updated scripts in package.json to account for new file locations and added more specific "name" attributes.

Not sure if there's any else we should add to the package.json files?

nathanrobb replied:
I think I'd potentially have the two apps run separately (i.e. not have anything about the client in the servers `package.json` and have the user start them both.

What you have is good though. No need for any more changes.

nathanrobb commented on client/public/index.html:
```diff
@@ -0,0 +1,20 @@
+      content="Web site created using create-react-app"
+    />
+    <link rel="apple-touch-icon" href="%PUBLIC_URL%/logo192.png" />
+    <link rel="manifest" href="%PUBLIC_URL%/manifest.json" />
```
Whats happening with this `%PUBLIC_URL%`? What does it do? Where is it set/pulled from?

mattegan111 replied:
This expects to be defined by the environment, which hasn't been set here in our development environment. When deploying to a service like Heroku this may change. If we wanted to specify a value in its place while still in development, we could set a "homepage" attribute in package.json.

Is this acceptable as is?

nathanrobb replied:
Ah, yup makes sense. Would using a relative URL work?
If not we could look to set a dev value to `localhost:port` if its not already. We ideally would have our dev environment mimic as closely as we can to our prod env. It reduces issues that only happen in production.

nathanrobb commented on package.json:
```diff
@@ -0,0 +1,36 @@
+{
+  "name": "fakebook",
+  "version": "1.0.0",
+  "description": "A facebook clone built using Node.js, React with hooks and Typescript. From The Odin Project.",
```
This project (and the client as well) isn't using typescript. It's using Javascript.

For express _maybe_ this doc would help? Or you can find another that you think would work, https://developer.okta.com/blog/2018/11/15/node-express-typescript

For Typescript with `create-react-app` have a look at https://create-react-app.dev/docs/adding-typescript/

nathanrobb commented on config/db.js:
```diff
@@ -0,0 +1,20 @@
+    .connect(db, {
+      useNewUrlParser: true,
+      useUnifiedTopology: true
+    })
```
We should prob add semicolons throughout our code, even though they aren't necessary, they help with readability. Up to you though. ESLint _should_ pick this up if it's configured for this project.

mattegan111 replied:
Done, using Prettier.

nathanrobb commented on config/db.js:
```diff
@@ -0,0 +1,20 @@
+const mongoose = require('mongoose')
+const config = require('config')
+const db = config.get('mongoURI')
```
Perhaps a better name for this would be `dbConnectionString` (or `dbConnStr` if you want it shorter) or `mongoUri`?

mattegan111 replied:
Renamed db to dbConnectionString.

mattegan111 committed:
Renamed db to dbConnectionString

Renamed variable to be more specific.

mattegan111 committed:
Move server files to dedicated folder

Created 'server' folder and added all files and folders relating to express. Updated scripts in package.json to account for new file locations and added more specific "name" attributes.

mattegan111 committed:
Update .gitignore

mattegan111 committed:
Reformat to include semicolons

Reformat using Prettier

nathanrobb reviewed (approved):
LGTM, on the assumption we're sticking with JS.

Maybe we should update the `package.json` to say JS instead of TS. Also, ideally our dev build would work as similar to the prod build as we can make it (re: the url comments).

These are minor changes and I'm not gonna suggest a change for these. You're welcome to do them though ;)

mattegan111 committed:
Remove mention of Typescript

No longer using Typescript for this project so corrected this description.

mattegan111 committed:
Adds logo images

These should not have been deleted after running create-react-app. We can replace them with a custom image in future if desired, and use these as defaults in the meantime.

mattegan111 committed:
Introduced relative links

Replaces use of %PUBLIC_URL% with relative links to make our development environment mimic the production environment more closely.
